app/controllers/teachers_controller.rb

```diff
@@ -20,41 +22,26 @@ def create
     @school = School.new(school_params)
     # Find by email, but allow updating other info.
     @teacher = Teacher.find_by(email: teacher_params[:email])
-    if @teacher
-      # Exclude website from the school finder, so we can update an existing school.
-      @school = School.find_or_create_by(name: school_params[:name], city: school_params[:city], state: school_params[:state])
-      @school.website = school_params[:website]
-      @school.save!
-      @teacher.update(teacher_params)
-      @teacher.school = @school
-      @teacher.save!
-      if @teacher.application_status == "Validated"
-        TeacherMailer.welcome_email(@teacher).deliver_now
-        TeacherMailer.form_submission(@teacher).deliver_now
-        TeacherMailer.teals_confirmation_email(@teacher).deliver_now
-        flash[:success] = "Thanks! We have updated your information. We have sent BJC info to #{@teacher.email}."
-      else
-        flash[:success] = "Thanks! We have updated your information."
-      end
+    if @teacher and defined?(current_user.id) and current_user.id == @teacher.id
+      params[:id] = current_user.id
+      update
+      return
+    end
+    @school = school_from_params
+    if !@school.save
+      flash[:alert] = "An error occured! #{@school.errors.full_messages}"
+      render 'new'
+      return
+    end
+    @teacher = @school.teachers.build(teacher_params)
```

Comment:
@christopher-hou Does this line raise ActiveRecord::RecordNotUnique if a used email is used?

Reply:
I think it's still just line 38 here (@teacher.save); most of the code in this part was there previously; I just removed a bunch of nested if statements to reduce cognitive complexity